test/test_datasets.py

```diff
@@ -29,6 +41,130 @@ def test_get_intersection_range(self):
+        self.assertIsInstance(content, bytes)
+
+        mock_get.return_value.status_code = 400
+        mock_get.return_value.content = bytes("WEATHER OR WILDFIRE FILE", 'utf-8')
+        self.assertRaises(requests.exceptions.ConnectionError, utils.url_retrieve, 'url')
+
+    def test_get_fname(self):
+
+        url_firms = "https://firms.modaps.eosdis.nasa.gov/data/active_fire/c6/csv/MODIS_C6_Europe_24h.csv"
+        url_ghcn = "https://www1.ncdc.noaa.gov/pub/data/ghcn/daily/by_year/2020.csv.gz"
+        url_isd = "https://www.ncei.noaa.gov/data/global-hourly/archive/csv/2020.tar.gz"
+
+        self.assertEqual(utils.get_fname(url_firms), ("MODIS_C6_Europe_24h", "csv", None))
+        self.assertEqual(utils.get_fname(url_ghcn), ("2020", "csv", "gz"))
+        self.assertEqual(utils.get_fname(url_isd), ("2020", None, "tar.gz"))
+
+    @staticmethod
+    def _make_tarfile():
+
+        unzipped_content = [
+            ['col1', 'col2', 'col3', 'col4'],
+            ['test', 'test', 'test', 'test'],
+            ['test', 'test', 'test', 'test'],
+            ['test', 'test', 'test', 'test']]
+
+        os.makedirs(os.path.dirname('./server/'), exist_ok=True)
+        with open("./server/test_tar.csv", mode='w') as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerows(unzipped_content)
+
+        out = tarfile.open('./server/test.tar.gz', 'w:gz')
+        out.add("./server/", arcname=os.path.basename("./server/"))
+        out.close()
+
+        with open('./server/test.tar.gz', 'rb') as tar_file:
+            memory_file = BytesIO(tar_file.read())
+
+        return memory_file
+
+    @staticmethod
+    def _make_gzipfile():
+
+        unzipped_content = [
+            ['col1', 'col2', 'col3', 'col4'],
+            ['test', 'test', 'test', 'test'],
+            ['test', 'test', 'test', 'test'],
+            ['test', 'test', 'test', 'test']]
+
+        os.makedirs(os.path.dirname('./server/'), exist_ok=True)
+        with open("./server/test_gz.csv", mode='w') as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerows(unzipped_content)
+
+        with gzip.GzipFile("./server/test.gz", mode='w') as gz, open("./server/test_gz.csv", mode='r') as csvfile:
+            gz.write(csvfile.read().encode())
+            gz.close()
+
+        with open('./server/test.gz', 'rb') as gz_file:
+            memory_file = BytesIO(gz_file.read())
+
+        return memory_file
+
+    @staticmethod
+    def _make_zipfile():
+
+        unzipped_content = [
+            ['col1', 'col2', 'col3', 'col4'],
+            ['test', 'test', 'test', 'test'],
+            ['test', 'test', 'test', 'test'],
+            ['test', 'test', 'test', 'test']]
+
+        os.makedirs(os.path.dirname('./server/'), exist_ok=True)
+        with open("./server/test_zip.csv", mode='w') as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerows(unzipped_content)
+
+        with ZipFile('./server/test.zip', 'w') as zip:
```

Comment:
`zip` is a built-in Python function, Codacy seems to dislike redefinition of those functions!
Since it's always temporary in the context, most of the time I use "file"

Reply:
Will do that